src/views/CreateRules.vue

```diff
@@ -6,17 +6,55 @@
 
 <script lang="ts">
 // Libraries
-import { Component, Vue } from 'vue-property-decorator'
+import { Component, Mixins, Vue, Watch } from 'vue-property-decorator'
+import { Action, Getter } from 'vuex-class'
+
+// Interfaces
+import { ActionBindingIF, CreateRulesIF } from '@/interfaces'
+
+// Mixins
+import { CommonMixin } from '@/mixins'
 
 // Components
 import { UploadRules } from '@/components/CreateRules'
 
+import { RouteNames } from '@/enums'
+
 @Component({
   components: {
     UploadRules
   }
 })
-export default class CreateRules extends Vue {}
+export default class CreateRules extends Mixins(CommonMixin) {
+  @Getter getShowErrors!: boolean
+  @Getter getCreateRulesStep!: CreateRulesIF
+
+  @Action setIgnoreChanges!: ActionBindingIF
+
+  /** Called when component is created. */
+  private created (): void {
+    // ignore data changes until page has loaded
+    this.setIgnoreChanges(true)
+    Vue.nextTick(() => {
+      this.setIgnoreChanges(false)
+    })
+  }
+
+  @Watch('$route')
```

Comment:
Just to confirm, this was ok? Only being called when you expected it to be etc

Reply:
yes, i believe this is ok and i've done a bunch of testing too.  The scrolling for each page level component should only happen when there are errors(`getShowErrors`) and the route matches the page level component.  Does that answer your question?

Reply:
For sure! 
I think Severin had concern it was called on any route change due to the way Vue hooks can create/mount components before they are displayed. If you see nothing abnormal than all good!

Reply:
yeah, it is probably ideal it is moved to `App` level but I moved it to at least the page level for now.  I made a note in the description.  I do think we want to eventually move it to the root component(`App`) level but it felt like a bigger change that would involve a bit more testing so I held off for now.  If we do it, I think we would have to do some refactoring of other components to match what the upload rules/memorandum components for tracking validation items.

@severinbeauvais are you ok with what i've done so far?  or would you prefer me to try and move everything to the `App` level?

Reply:
My initial thought was to have a single route change handler, but actually I think what you've done here is OK: you have a route handler on this page, and another one on the Create Memorandum page. That keeps things localized, which is good.

Less good is that this page (all pages in fact) are always rendered/instantiated, but conditionally shown - see App.vue template. (This is not true of some components, which are conditionally rendered per v-if directives.) This means all route change handlers always run. I guess you haven't seen any issues with this but it might be good to confirm this, and maybe have the handlers of hidden pages do nothing (check "if shown"?).

Reply:
Oh my, I see you've already done this (checking the route)... well done! Sorry, I need to look closer :)

Reply:
So - I'm totally OK with this.